### PR TITLE
Adding code that trains on different LLM models and added the introduction to GLayout notebook

### PR DESCRIPTION
**Pull request by srpathen**
1. We train on phi, mistral, and codestral
2. Added introduction to GLayout notebook that introduces many of the functions that the participants will use
3. Edited workflow scripts to downgrade numpy to version 1.23.3
4. Added a convo parser

### CODE REVIEW

**Review by msaligane — APPROVED** [review]

**Comment by msaligane** [comment]
The CI is failing for these. Is this normal? @chetanyagoyal and @srpathen

**Comment by chetanyagoyal** [comment]
The CI failing is unintended behaviour, we will look into this

File: .github/workflows/glayout_opamp_sim.yml [hunk 1]
```diff
@@ -41,6 +41,8 @@ jobs:
                       pip3 install -r requirements.txt &&\
                       cd ./openfasoc/generators/glayout/tapeout/tapeout_and_RL/ &&\
                       pip3 install prettyprint prettyprinttree gdstk &&\
+                      python3 -m pip uninstall numpy -y &&\
+                      python3 -m pip install numpy==1.23.3 &&\
```
Comment (chetanyagoyal) [inline .github/workflows/glayout_opamp_sim.yml:45]: can this be implemented in the requirements.txt file instead, pin the version there

File: .github/workflows/glayout_sky130.yml [hunk 2]
```diff
@@ -41,6 +41,8 @@ jobs:
                     cp ./.github/scripts/test_glayout_ci.py ./openfasoc/generators/glayout/. &&\
                     cd ./openfasoc/generators/glayout &&\
                     pip3 install gdsfactory==7.7.0 prettyprint prettyprinttree gdstk &&\
+                    python3 -m pip uninstall numpy -y &&\
+                    python3 -m pip install numpy==1.23.3 &&\
```
Comment (chetanyagoyal) [inline .github/workflows/glayout_sky130.yml:45]: same here

File: .github/workflows/glayout_sky130.yml [hunk 3]
```diff
@@ -62,6 +64,8 @@ jobs:
...
```
Comment (chetanyagoyal) [inline .github/workflows/glayout_sky130.yml:68]: same here

File: .github/workflows/glayout_sky130.yml [hunk 4]
```diff
@@ -82,6 +86,8 @@ jobs:
...
```
Comment (chetanyagoyal) [inline .github/workflows/glayout_sky130.yml:90]: same here

**Review by chetanyagoyal — CHANGES_REQUESTED** [review]

**Comment by chetanyagoyal** [comment]
Glayout changes are okay, they are previously unmerged changes to `llm`

**Comment by chetanyagoyal** [comment]
@srpathen scipy conflicts with the numpy version you're installing, we need numpy==1.23.5 or greater

**Comment by srpathen** [comment]
Updated the numpy version to 1.23.5

**Comment by msaligane** [comment]
Is this ready for a second review?

**Comment by chetanyagoyal** [comment]
@msaligane this can be merged

**Review by msaligane — APPROVED** [review]

**Pull request merged**